app/controllers/products_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ProductsController < ApplicationController
+  before_action :find_category
+  before_action :fetch_products, only: [:index]
+  def index
+    if params[:query].present?
+      @products = @products.search_by_name(params[:query])
+    end
+    if params[:price_filter].present?
+      @products = filter_by_price(@products, params[:price_filter])
+    end
+    @pagy, @paginated_products = pagy @products, items: Settings.digist.digist_10
+    @total_count = @products.size
+  end
+
+  private
+  def find_category
+    @category = Category.find_by(id: params[:category_id])
+  end
+  def fetch_products
+    @products = if @category
+                  @category.product
+                else
+                  Product.all
+                end
```

Comment:
```suggestion
    @products = @category ? @category.product : Product.all
```
Suggest thế chứ chỗ này nên viết scope nhé.

Reply:
done ạ